lifecycle/createcommon.go

```
@@ -72,8 +85,160 @@ func (rbc *ReleaseBundleCreateCommand) Run() error {
+		}
+		detectedCreationSources = append(detectedCreationSources, sourceType)
+	}
+
+	if err := validateCreationSources(detectedCreationSources); err != nil {
+		return "", err
+	}
+	return detectedCreationSources[0], nil
+}
+
+func validateCreationSources(detectedCreationSources []services.SourceType) error {
+	if len(detectedCreationSources) == 0 {
+		return errorutils.CheckErrorf("unexpected err while validating spec - could not detect any creation sources")
+	}
+
+	// Assert single creation source.
+	for i := 1; i < len(detectedCreationSources); i++ {
+		if detectedCreationSources[i] != detectedCreationSources[0] {
+			return generateSingleCreationSourceErr(detectedCreationSources)
+		}
+	}
+
+	// If aql, assert single file.
+	if detectedCreationSources[0] == services.Aql && len(detectedCreationSources) > 1 {
+		return errorutils.CheckErrorf("only a single aql query can be provided")
+	}
+	return nil
+}
+
+func generateSingleCreationSourceErr(detectedCreationSources []services.SourceType) error {
+	var detectedStr []string
+	for _, source := range detectedCreationSources {
+		detectedStr = append(detectedStr, string(source))
+	}
+	return errorutils.CheckErrorf(
+		"multiple creation sources were detected in separate spec files. Only a single creation source should be provided. Detected: '%s'",
+		coreutils.ListToText(detectedStr))
+}
+
+func validateFile(file spec.File) (services.SourceType, error) {
+	// Aql creation source:
+	isAql := len(file.Aql.ItemsFind) > 0
+
+	// Build creation source:
+	isBuild := len(file.Build) > 0
+	isIncludeDeps, _ := file.IsIncludeDeps(false)
+
+	// Bundle creation source:
+	isBundle := len(file.Bundle) > 0
+
+	// Build & bundle:
+	isProject := len(file.Project) > 0
+
+	// Artifacts creation source:
+	isPattern := len(file.Pattern) > 0
+	isExclusions := len(file.Exclusions) > 0 && len(file.Exclusions[0]) > 0
+	isProps := len(file.Props) > 0
+	isExcludeProps := len(file.ExcludeProps) > 0
+	isRecursive, _ := file.IsRecursive(true)
```

Let's not ignore the error. It should fail if we provide `--recursive=bla`